packages/form-core/tests/FieldApi.spec.ts

```diff
@@ -932,6 +932,101 @@ describe('field api', () => {
     })
   })
 
+  it('should run listener onChange', () => {
+    const form = new FormApi({
+      defaultValues: {
+        name: 'test',
+      },
+    })
+
+    let triggered!: string
+    const field = new FieldApi({
+      form,
+      name: 'name',
+      listeners: {
+        onChange: ({ value }) => {
+          triggered = value
+        },
+      },
+    })
+
+    field.mount()
+
+    field.setValue('other')
+    expect(triggered).toStrictEqual('other')
+  })
+
+  it('should run listener onBlur', () => {
+    const form = new FormApi({
+      defaultValues: {
+        name: 'test',
+      },
+    })
+
+    let triggered!: string
+    const field = new FieldApi({
+      form,
+      name: 'name',
+      listeners: {
+        onBlur: ({ value }) => {
+          triggered = value
+        },
+      },
+    })
+
+    field.mount()
+
+    field.handleBlur()
+    expect(triggered).toStrictEqual('test')
+  })
+
+  it('should run listener onMount', () => {
+    const form = new FormApi({
+      defaultValues: {
+        name: 'test',
+      },
+    })
+
+    let triggered!: string
+    const field = new FieldApi({
+      form,
+      name: 'name',
+      listeners: {
+        onMount: ({ value }) => {
+          triggered = value
+        },
+      },
+    })
+
+    field.mount()
+
+    expect(triggered).toStrictEqual('test')
+  })
+
+  it('should run listener onSubmit', () => {
+    const form = new FormApi({
+      defaultValues: {
+        name: 'test',
+      },
+    })
+
+    let triggered!: string
+    const field = new FieldApi({
+      form,
+      name: 'name',
+      listeners: {
+        onSubmit: ({ value }) => {
+          triggered = value
+        },
+      },
+    })
+
+    field.mount()
+    form.handleSubmit()
+
+    expect(triggered).toStrictEqual('test')
+  })
+
```

Comment:
I created the test before the functionality is implemented (looking for guidance on implementation) , so this will naturally fail.

Reply:
Moved test to FormApi.spec.ts since my current solution lives there.